warehouse/retry.go

```diff
@@ -122,6 +177,35 @@ func (retryReq *RetryRequest) retryUploads(sourceIDs []string) (rowsAffected int
 	return
 }
 
+func (retryReq *RetryRequest) countUploadsToRetry(ctx context.Context, sourceIDs []string) (count int64, err error) {
+	var clausesQuery string
+	var clauses []string
+	var clausesArgs []interface{}
+
+	// Preparing clauses query
+	clauses, clausesArgs = retryReq.whereClauses(sourceIDs)
+	for i, clause := range clauses {
+		clausesQuery = clausesQuery + strings.Replace(clause, retryQueryPlaceHolder, fmt.Sprintf("$%d", i+1), 1)
+		if i != len(clauses)-1 {
+			clausesQuery += " AND "
+		}
+	}
+
+	// Preparing the prepared statement
+	preparedStatement := fmt.Sprintf(`
+		SELECT
+		  count(*)
+		FROM
+		  wh_uploads
+		WHERE %[1]s`,
+		clausesQuery,
+	)
+
+	// Executing the statement
+	err = retryReq.API.dbHandle.QueryRowContext(ctx, preparedStatement, clausesArgs...).Scan(&count)
```

Comment:
I would refactor this query to the dbModule we created and call that over the retryReq.API.dbHandle.GetUploadsCount(...filterClause string). Wdyt ?

It will allow us to have a generic function call to count the entries in the wh_uploads with filter params and can be used by other candidates as well.

Reply:
Thanks for pointing this out. It's a very good motivation to start moving the Queries to DB module.

Reply:
Done.